### PR TITLE
Stage aayushi

### PR DESCRIPTION
**Pull request by Aayushitmdc**
This pull request contains:
- some language fixes, 
- The lens source has been added to the index page navigation.
- corrected links which failed in link checking.

### CODE REVIEW

File: docs/products/data_product/how_to_guides/index.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ To know more about each phase, please explore below sections.
...
+The Design phase of the Data Product life-cycle is pivotal in aligning business objectives with actionable solutions. It begins with a comprehensive understanding of business goals and use cases, forming the basis for developing a robust solution architecture. DataOS resources required to design a Data Product are [Instance Secret](/resources/instance_secret/), [Depot](/resources/depot/), [Cluster](/resources/cluster/), and [Stacks](/resources/stacks/). DataOS also provides interfaces that help to design the Data Product such as [Data Product Hub](/interfaces/data_product_hub) where you can explore existing Data Products, [Workbench](/interfaces/workbench/) to query your data source without data movement, and [Metis](/interfaces/metis) to get insight about the meta data. To know more in detail, please refer to  [How to Design a Data Product](/products/data_product/how_to_guides/design/).
```
Comment (pyjs-tmdc) [inline docs/products/data_product/how_to_guides/index.md:8]: The link for the data product hub and workbench need correction

File: docs/resources/lens/data_sources.md [hunk 2]
```diff
@@ -11,3 +11,7 @@
 [Postgres](/resources/lens/data_sources/postgres/)
 
 [Minerva](/resources/lens/data_sources/minerva/)
+
+[Flash](/resources/lens/data_sources/flash/)
```
Comment (pyjs-tmdc) [inline docs/resources/lens/data_sources.md:15]: why have you commented Snowflake

File: docs/resources/lens/index.md [hunk 3]
```diff
@@ -137,6 +137,12 @@ After creating a Lens data model, the next step is to explore it—this means in
   <!-- - [Supported Parameters for Tables and Views](/resources/lens/supported_parameters_for_tables_and_views/)
     Understand the role of logical tables and views in data models. -->
 
+## Supported data sources
+
+The Lens can be integrated with a variety of data sources. This section outlines the data sources supported by Lens and provides insights into connecting and managing data to ensure optimal performance and scalability.
+
+[Supported Data Sources](/resources/lens/data_sources/)
```
Comment (pyjs-tmdc) [inline docs/resources/lens/index.md:144]: You have to add the link of all the pages on the main page of Lens itself of all the supported sources, not create a sub page for it

**Review by pyjs-tmdc — CHANGES_REQUESTED** [review]
@Aayushitmdc I have put some comments on the PR that you had created.

File: docs/products/data_product/how_to_guides/index.md (hunk 1 above)
Comment (Aayushitmdc) [reply docs/products/data_product/how_to_guides/index.md:8]: Done

**Review by pyjs-tmdc — APPROVED** [review]

**Pull request merged**